app/registries/foreman/plugin/importer_registry.rb

```diff
@@ -0,0 +1,23 @@
+module Foreman
+  class Plugin
+    class ImporterRegistry
+      attr_accessor :importers 
+
+      def initialize
+        @importers = {
+          :puppet => 'PuppetFactImporter'
+        }.with_indifferent_access
```

Comment:
`with_indifferent_access` may not be necessary. It looks only the `register` method accesses the default via dot-notation to set a default, which could be changed and not require indifferent access.

Reply:
We can skip the `indifferent_access` only if we agree on the type of keys - either `:puppet` or `'puppet'`. The problem is that from code it's better to work with symbols (`registry.get(:puppet)`), but  the request uses string as parameters.